api-service/cloudify_api/listener.py

```diff
@@ -0,0 +1,70 @@
+import asyncio
+import logging
+import json
+
+import asyncpg
+
+
+class ListenerException(Exception):
+    pass
+
+
+class TaskAlreadyExists(ListenerException):
+    def __init__(self, channel: str):
+        self.channel = channel
+
+
+class TaskNotDefined(ListenerException):
+    def __init__(self, channel: str):
+        self.channel = channel
+
+
+class Listener:
+    def __init__(self, dsn: str, logger: logging.Logger):
+        self.dsn = dsn
+        self.logger = logger
+        self.conn_listen = None
+        self.loop = asyncio.get_event_loop()
+        self.channels = {}
+
+    def listen_on_channel(self, channel: str):
+        if channel in self.channels.keys():
+            raise TaskAlreadyExists(channel)
+        self.channels[channel] = {
+            'task': self.loop.create_task(self._listener(channel)),
+            'queues': [],
```

Comment:
I'd make this a set, for faster .remove() (or .discard) if there's thousands of connected clients

Reply:
Good point.  Done.  I left the `.remove()` because I think that might be interesting to log such situations.